### PR TITLE
3.1 All products shouldn't be visible for all staff users 

### PR DESCRIPTION
**Pull request by CossackDex**
I want to merge this change because...

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-4302 All products shouldn't be visible for all staff users.](https://app.clickup.com/t/2549495/SALEOR-4302)

File: saleor/graphql/page/tests/test_page.py [hunk 1]
```diff
@@ -113,21 +113,68 @@ def test_customer_query_unpublished_page(user_api_client, page):
...
+    # query by ID
+    variables = {"id": graphene.Node.to_global_id("Page", page.id)}
+    response = staff_api_client.post_graphql(
+        PAGE_QUERY,
+        variables,
+        permissions=[permission_manage_pages],
+        check_no_permissions=False,
+    )
+    content = get_graphql_content(response)
+    assert content["data"]["page"] is not None
```
Comment (IKarbowiak) [inline saleor/graphql/page/tests/test_page.py:131]: IMO we should check if the return data is the page that we expected.

File: saleor/graphql/page/tests/test_page.py [hunk 2]
```diff
@@ -113,21 +113,68 @@ def test_customer_query_unpublished_page(user_api_client, page):
...
+    # query by slug
+    variables = {"slug": page.slug}
+    response = staff_api_client.post_graphql(
+        PAGE_QUERY,
+        variables,
+        permissions=[permission_manage_pages],
+        check_no_permissions=False,
+    )
+    content = get_graphql_content(response)
     assert content["data"]["page"] is not None
```
Comment (IKarbowiak) [inline saleor/graphql/page/tests/test_page.py:163]: Same here

**Review by IKarbowiak — APPROVED** [review]

File: saleor/graphql/page/tests/test_page.py (hunk 1 above)
Comment (CossackDex) [reply saleor/graphql/page/tests/test_page.py:131]: @korycins

Comment (korycins) [reply saleor/graphql/page/tests/test_page.py:131]: Looks like a good suggestion. I just copy and paste the test and modify them to check the scenario. But for changes to master, we can extend them as Iga propsed.

**Review by IKarbowiak — APPROVED** [review]

File: saleor/core/permissions.py [hunk 3]
```diff
@@ -158,3 +159,25 @@ def get_permissions_from_codenames(permission_codenames: List[str]):
         .prefetch_related("content_type")
         .order_by("codename")
     )
+
+
+def permission_required(perms, context):
```
Comment (korycins) [inline saleor/core/permissions.py:164]: It is not a `context` it is a requestor.

**Review by korycins — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**